test/unit/model/authorization/TestTakerAuthorizationServiceTest.php

```diff
@@ -0,0 +1,300 @@
+            ->willReturn($property);
+
+        $delivery->expects($this->once())
+            ->method('getOnePropertyValue')
+            ->with($property)
+            ->willReturn($propertyValue);
+
+        $deliverySyncServiceMock->method('isProctoredByDefault')->willReturn($proctorByDefault);
+
+        $this->service->setServiceLocator(
+            $this->getServiceLocatorMock(
+                [
+                    Ontology::SERVICE_ID => $this->ontologyMock,
+                    DeliverySyncService::SERVICE_ID => $deliverySyncServiceMock
+                ]
+            )
+        );
+
+        if ($expected) {
+            $this->assertTrue(
+                $this->service->isProctored('deliveryUri', $this->getMock(User::class))
+            );
+        } else {
+            $this->assertFalse(
+                $this->service->isProctored('deliveryUri', $this->getMock(User::class))
+            );
+        }
+    }
+
+    /**
+     * @return array
+     */
+    public function isProctoredDataProvider()
+    {
+        return [
+            'byDefault' => [null, true, true],
+            'byDefaultNo' => [null, false, false],
+            'proctored' => [
+                new core_kernel_classes_Property('http://www.tao.lu/Ontologies/TAODelivery.rdf#ComplyEnabled'),
+                false,
+                true
+            ],
+            'notProctored' => [
+                new core_kernel_classes_Property('http://www.tao.lu/Ontologies/TAODelivery.rdf#ComplyDisabled'),
+                true,
+                false
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider verifyResumeAuthorizationDataProvider
+     * @param string $state
+     * @param bool $isSuccess
+     * @throws Exception
+     */
+    public function testVerifyResumeAuthorization($state, $isSuccess)
+    {
+        $this->ontologyMock->method('getProperty')
+            ->willReturnOnConsecutiveCalls(
+                new core_kernel_classes_Property('http://www.tao.lu/Ontologies/TAODelivery.rdf#ComplyEnabled'),
+                new core_kernel_classes_Property(
+                    'http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryTestRunnerFeatures'
+                )
+            );
+
+        $delivery = $this->getDeliveryMock();
+        $delivery->method('getOnePropertyValue')->willReturnOnConsecutiveCalls(
+            new core_kernel_classes_Property('http://www.tao.lu/Ontologies/TAODelivery.rdf#ComplyEnabled'),
+            'feature'
+        );
+
+        $deliveryExecutionMock = $this->getDeliveryExecutionMock($state);
+        $deliveryExecutionMock->method('getDelivery')->willReturn($delivery);
+
+        $this->service->setServiceLocator(
+            $this->getServiceLocatorMock(
+                [
+                    Ontology::SERVICE_ID => $this->ontologyMock,
+                    DeliverySyncService::SERVICE_ID => $this->getMock(DeliverySyncService::class)
+                ]
+            )
+        );
+        if (!$isSuccess) {
```

Comment:
If you test than method throws exception it's better to have it as a separate test case where you will test all possible scenarios with exception.

Reply:
fixed